version: '3.7'

services:
  db:
    image: postgres:11-alpine
    environment:
      POSTGRES_USER: ${POSTGRES_USER:-postgres}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD:-123456}
      PGDATA: /data/postgres
      POSTGRES_DB: shop_db
    ports:
      - "5432:5432"
    networks:
      - postgres
    restart: "on-failure"

  pgadmin:
    container_name: pgadmin_container
    image: dpage/pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: ${PGADMIN_DEFAULT_EMAIL:-pgadmin4@pgadmin.org}
      PGADMIN_DEFAULT_PASSWORD: ${PGADMIN_DEFAULT_PASSWORD:-admin}
      PGADMIN_CONFIG_SERVER_MODE: 'False'

    ports:
      - "5050:80"
    networks:
      - postgres
    restart: unless-stopped

  redis:
    image: "redis:alpine"
  app:
      image: "python:3"
      volumes:
      - .:/shop
      ports:
      - 8000:8000
      command: bash -c "cd shop && pip install -r requirements.txt && python manage.py makemigrations && python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
      depends_on:
        - db
      networks:
        - postgres
networks:
  postgres:
    driver: bridge